jobs/metrics-report-generator/src/utils/helpers.util.ts

```diff
@@ -0,0 +1,98 @@
+import { logError, logInfo, logWarn } from '@interop-be-reports/commons'
+import {
+  AgreementsWorksheetTableData,
+  DescriptorsWorksheetTableData,
+  TokensWorksheetTableData,
+} from '../models/excel.model.js'
+import { AgreementQueryData, EServiceQueryData, PurposeQueryData, TenantQueryData } from '../models/query-data.model.js'
+import { randomUUID } from 'crypto'
+import { TokensDataQueryResult } from '../services/athena-queries.service.js'
+
+const cidJob = randomUUID()
+
+export const log = {
+  info: logInfo.bind(null, cidJob),
+  warn: logWarn.bind(null, cidJob),
+  error: logError.bind(null, cidJob),
+}
+
+export function generateAgreementsWorksheetTableData(
+  agreements: AgreementQueryData[],
+  purposes: PurposeQueryData[],
+  eservicesMap: Map<string, EServiceQueryData>,
+  tenantsMap: Map<string, TenantQueryData>
+): AgreementsWorksheetTableData[] {
+  return agreements.map<AgreementsWorksheetTableData>((agreement) => {
+    const agreementPurposes = purposes.filter((purpose) => purpose.eserviceId === agreement.eserviceId)
+    const consumer = tenantsMap.get(agreement.consumerId)
+    const producer = tenantsMap.get(agreement.producerId)
+    const eservice = eservicesMap.get(agreement.eserviceId)
+
+    if (!consumer) log.warn(`[Agreements Worksheet] Tenant consumer ${agreement.consumerId} not found in readmodel`)
+    if (!producer) log.warn(`[Agreements Worksheet] Tenant producer ${agreement.producerId} not found in readmodel`)
+    if (!eservice) log.warn(`[Agreements Worksheet] Eservice ${agreement.eserviceId} not found in readmodel`)
+
+    return AgreementsWorksheetTableData.parse({
```

Comment:
is it required to parse data at runtime instead of using compile time check?
(same for other occurrences)

Reply:
As I said in the other PR, you're right, this is not needed here (and probably not needed also in most of the other occurencies)